ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committing/ProcessTransactionSubmission.scala

```diff
@@ -86,13 +86,20 @@ private[kvutils] class ProcessTransactionSubmission(
   ): Commit[Unit] = {
     val dedupKey = commandDedupKey(transactionEntry.submitterInfo)
     get(dedupKey).flatMap { dedupEntry =>
-      if (dedupEntry.isEmpty) {
+      def isAfterDeduplicationTime(stateValue: DamlStateValue): Boolean = {
```

Comment:
Please move this function out of here to improve readability and make it testable.

Reply:
Fixed. See latest commit.